darts/timeseries.py

```diff
@@ -304,6 +304,30 @@ def copy(self, deep: bool = True) -> 'TimeSeries':
         else:
             return TimeSeries(self._df, self.freq_str())
 
+    def split(self, proportion: float) -> Tuple['TimeSeries', 'TimeSeries']:
+        """
+        Splits he TimeSeries in two, based on a given proportion.
```

Comment:
I think here you should specify what is the boundary behavior. What happens when I call `series.split(0.5)` and `series` has length 5? Do I get back 2 + 3 or 3 + 2?
In fact you could perhaps just augment the signature of `split_before()` and `split_after()` in order to support float (and int - see other comment below).

Reply:
I think augmenting the signature of `split_before()` and `split_after()` is a really good idea. It is also a good idea to use the function `utils.get_timestamp_at_point()` to stick to DRY.  In order to use it, it was necessary to move `utils.get_timestamp_at_point()` to be a method of `TimeSeries` to avoid circular dependencies, so I had to do some refactoring.

Let me know if you agree with this approach. I am still hesitating if the `integer` passed should be an index or the number of samples

Reply:
> I think augmenting the signature of `split_before()` and `split_after()` is a really good idea. It is also a good idea to use the function `utils.get_timestamp_at_point()` to stick to DRY. In order to use it, it was necessary to move `utils.get_timestamp_at_point()` to be a method of `TimeSeries` to avoid circular dependencies, so I had to do some refactoring.
> 
> Let me know if you agree with this approach. I am still hesitating if the `integer` passed should be an index or the number of samples

I would vote for an index. Number of points might make sense in some context, but I think it might be confusing if it returns something different than the corresponding `__getitem__()`